### ISSUE TITLE
Geospatial PDF

### ISSUE THREAD

**Issue by nempoBu4** [Kozea/WeasyPrint#1137]
Is it possible to create [geospatial PDF](https://helpx.adobe.com/sea/acrobat/using/geospatial-pdfs.html) with Weasyprint?

**Comment by liZe** [comment]
It’s probably possible, if you know how to add the geospatial metadata the PDF file after it’s generated by Cairo.

You can try the [`finisher` feature](https://github.com/Kozea/WeasyPrint/pull/1096) recently added to the master branch if you are a PDF master!

**Comment by grewn0uille** [comment]
Hello!

If you’re interested in geospatial PDF, we created a [short survey](https://app.evalandgo.com/s/index.php?id=JTk4cCU5NG8lOUQlQUQ%3D&a=JTk3biU5M2glOUMlQUI%3D) where you can give a boost to this feature and help us to improve WeasyPrint 😉

Vote for it!

**Comment by liZe** [comment]
@nempoBu4 We now support PDF/A-4e that replaces PDF/E. Would that be useful for you?

**Comment by nempoBu4** [comment]
@liZe, thank you for your work. I don't know how it will be useful for us. We have implemented Geospatial PDF support in our fork of WeasyPrint. This was enough to cover all our needs.

**Comment by liZe** [comment]
> [@liZe](https://github.com/liZe), thank you for your work. I don't know how it will be useful for us. We have implemented Geospatial PDF support in our fork of WeasyPrint. This was enough to cover all our needs.

Thanks for your answer.

Is your fork public? It could help to understand your needs, and maybe upstream these features!

**Comment by nempoBu4** [comment]
@liZe, yes it's [public](https://github.com/kagis/WeasyPrint).  You can take a look to our changes [here](https://github.com/Kozea/WeasyPrint/compare/main...kagis:WeasyPrint:geopdf).

**Comment by liZe** [comment]
It should be possible to do this in a finisher function, and if it’s not we should add a way to make it possible. Let’s keep this issue open until we find a solution to support extra data.

Related to #2583.
